python/GafferSceneUI/SelectionToolUI.py

```diff
@@ -112,3 +106,78 @@ class _RightSpacer( GafferUI.Spacer ) :
 	def __init__( self, imageView, **kw ) :
 
 		GafferUI.Spacer.__init__( self, size = imath.V2i( 0, 0 ) )
+
+class SelectModePlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plugs, **kw ) :
+
+		self.__menuButton = GafferUI.MenuButton( "", menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ) ) )
+
+		GafferUI.PlugValueWidget.__init__( self, self.__menuButton, plugs, **kw )
+
+	def _updateFromValues( self, values, exception ) :
+
+		if exception is not None :
+			self.__menuButton.setText( "" )
+		else :
+			modifiers = dict(
+				zip(
+					GafferSceneUI.SelectionTool.registeredSelectModes(),
+					GafferSceneUI.SelectionTool.registeredSelectModeLabels()
+				)
+			)
+
+			assert( len( values ) == 1 )
+			label = modifiers.get( values[0], None )
+
+			if label is not None :
+				self.__menuButton.setText( "Any" if label == "Any" else label.partition( "/" )[-1] )
+			else :
+				self.__menuButton.setText( "Invalid" )
+
+		self.__menuButton.setErrored( exception is not None )
+
+	def _updateFromEditable( self ) :
+
+		self.__menuButton.setEnabled( self._editable() )
+
+	def __menuDefinition( self ) :
+
+		result = IECore.MenuDefinition()
+
+		modifiers = dict(
+			zip(
+				GafferSceneUI.SelectionTool.registeredSelectModes(),
+				GafferSceneUI.SelectionTool.registeredSelectModeLabels()
+			)
+		)
+
+		modifierCategories = set()
+
+		with self.getContext() :
+			currentValue = self.getPlug().getValue()
+
+		for modifierName, modifierLabel in modifiers.items() :
+			category, sep, label = modifierLabel.partition( "/" )
+
+			if category == "Any" :
+				label = "Any"
+
+			if category != "Any" and category not in modifierCategories :
+				result.append( f"/__{category}Dividier", { "divider" : True, "label" : category } )
+				modifierCategories.add( category )
```

Comment:
Probably unimportant, but there's an assumption here that all the things in one category will be grouped together in `registeredSelectModes()` - if not, then things end up under the wrong category. Probably fixable using `insertAfter()`, and using `modifiedCategories` to map to the last item in each category.

Like I say, unimportant. The only reason I mention is that I quite like this presentation, so I'm musing about it being an optional mode for PresetsPlugValueWidget, or even Menu. For that we'd need to be more robust I think.

Reply:
May as well fix it now before someone gets frustrated trying to do something clever with the menu. I think I have a good solution in https://github.com/GafferHQ/gaffer/pull/5720/commits/00a56aa5e1225494185a1219867516da4ac76664